File: string/0788_rotated_digits/0788_rotated_digits.py
class Solution:
    def rotatedDigits(self, N: int) -> int:
        ans = 0
        # For each x in [1, N], check whether it's good
        for x in range(1, N+1):
            S = str(x)
            # Each x has only rotateable digits, and one of them
            # rotates to a different digit
            ans += (all(d not in '347' for d in S)
                    and any(d in '2569' for d in S))
        return ans


N = 10
res = Solution().rotatedDigits(N)
print(res)

File: string/0788_rotated_digits/0788_rotated_digits.go
package main

import "fmt"

func good(n int, flag bool) bool {
	if n == 0 {
		return flag
	}
	d := n % 10
	switch d {
	case 3, 4, 7:
		return false
	case 0, 1, 8:
		return good(n/10, flag)
	}
	//2,5,6,9
	return good(n/10, true)
}

func rotatedDigits(N int) int {
	flag := false
	ans := 0
	for i := 1; i <= N; i++ {
		if good(i, flag) {
			ans++
		}
	}
	return ans
}

func main() {
	N := 10
	res := rotatedDigits(N)
	fmt.Println(res)
}
